### PR TITLE
Add docstrings to exported modules

### LINKED ISSUE #23959: Each module should have a README
```julia
julia> using DelimitedFiles

help?> DelimitedFiles
search: DelimitedFiles

  No documentation found.

  No docstring or README.md found for module DelimitedFiles.
```

### PR DESCRIPTION
**Pull request by jw3126**
This PR at least adds stubs and a test that docstrings exist. 
~~I failed to add a docstring to Base itself, how do I do that?~~
fix #23959

### CODE REVIEW

File: base/docs/basedocs.jl [hunk 1]
```diff
@@ -1492,4 +1492,8 @@ See the manual section on [Tuple Types](@ref).
 """
 Tuple
 
+@doc """
+The standard library of Julia.
+""" -> Base
```
Comment (mschauer) [inline base/docs/basedocs.jl]: As there is/will be a  "standard library", see https://github.com/JuliaLang/julia/tree/master/stdlib, it Base should not be explained as standard library.

Comment (jw3126) [reply base/docs/basedocs.jl]: Good point. At the moment `?Base` does not anyway. Any idea how to fix that?

Comment (mschauer) [reply base/docs/basedocs.jl]: You could document it as keyword
```
 @doc """
 bla bla
 """ -> kw"Base"
```

File: base/deprecated.jl [hunk 2]
```diff
@@ -1355,6 +1362,15 @@ module DSP
 end
 using .DSP
 export conv, conv2, deconv, filt, filt!, xcorr
+@doc """
+DFT will be removed from Base. Use the package with the same name instead.
```
Comment (ararslan) [inline base/deprecated.jl]: There's no package called DFT. The functionality has been moved to AbstractFFTs.

File: base/deprecated.jl [hunk 3]
```diff
@@ -709,8 +709,12 @@ module Collections
         @eval Base.@deprecate_moved $f "DataStructures"
     end
 end
+@doc """
+Collections will be removed from Base. Use DataStructures instead.
```
Comment (fredrikekre) [inline base/deprecated.jl]: This is part of the 0.6 deprecation, so will never see the light of day.

File: base/deprecated.jl [hunk 4]
```diff
@@ -1355,6 +1362,15 @@ module DSP
...
 export conv, conv2, deconv, filt, filt!, xcorr
+@doc """
+DFT will be removed from Base. Use `AbstractFFTs` instead.
+""" -> DFT
+@doc """
+DSP will be removed from Base. Use the package with the same name instead.
+""" -> DSP
+@doc """
+FFTW will be removed from Base. Use the package with the same name instead.
+""" -> FFTW
```
Comment (fredrikekre) [inline base/deprecated.jl]: Should we bother documenting these moved modules?

File: base/distributed/Distributed.jl [hunk 5]
```diff
@@ -1,5 +1,8 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+"""
+Tools for distributed parallel calculations.
```
Comment (fredrikekre) [inline base/distributed/Distributed.jl]: Perhaps change calculations to processing?

File: base/docs/basedocs.jl [hunk 6]
```diff
@@ -1492,4 +1492,8 @@ See the manual section on [Tuple Types](@ref).
 """
 Tuple
 
+@doc """
+The base library of Julia.
+""" -> kw"Base"
```
Comment (fredrikekre) [inline base/docs/basedocs.jl]: This can be:
```julia
"""
The base library of Julia.
"""
Base
```
instead of documenting as a keyword.

**Review by fredrikekre — COMMENTED** [review]
I guess these should be added in `doc/src/stdlib` somewhere to show up in the docs too. We can just have a modules section that list these.

File: base/docs/basedocs.jl (hunk 6 above)
Comment (jw3126) [reply base/docs/basedocs.jl]: Are you sure? I tried 
```Julia
@doc "..." -> Base
```
and `?Base` did not work.

File: base/deprecated.jl (hunk 4 above)
Comment (jw3126) [reply base/deprecated.jl]: I have a test that all exported modules in Base are documented. Thats why I added these.

File: base/docs/basedocs.jl (hunk 6 above)
Comment (jw3126) [reply base/docs/basedocs.jl]: I just tested your suggestion, it does not work either.

Comment (fredrikekre) [reply base/docs/basedocs.jl]: Hmm... I swear it worked yesterday when I tried :)

File: base/deprecated.jl (hunk 3 above)
Comment (jw3126) [reply base/deprecated.jl]: See below.

File: base/deprecated.jl (hunk 4 above)
Comment (fredrikekre) [reply base/deprecated.jl]: Yea, I'm not sure we need a test like that. Adding a new exported module in Base does not happen very often so not sure its worth it. If you add a `## Base Modules` section in `doc/src/stdlib/base.md` (perhaps just after the `## Keywords` section) and list the new docstrings there, then CI will fail if these docstrings are lost.

Comment (jw3126) [reply base/deprecated.jl]: Yes I agree the test was useful for making sure that I did not forget any exported module, but is useless now. I will add the Modules to `base.md`.

File: base/linalg/blas.jl [hunk 7]
```diff
@@ -1,6 +1,9 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 module BLAS
+@doc """
+Interface to BLAS subroutines.
+""" -> BLAS
```
Comment (jw3126) [inline base/linalg/blas.jl]: If I do 
```julia
"""
Interface to BLAS subroutines.
"""
BLAS
```
it won't show up in `?BLAS`. Any idea why?

**Comment by jw3126** [comment]
Does this look good now?

**Review by fredrikekre — APPROVED** [review]
Thanks. The docstrings can always be improved later if need be.

**Pull request merged**